go/ql/lib/semmle/go/frameworks/RsCors.qll

```diff
@@ -0,0 +1,166 @@
+/**
+ * Provides classes for modeling the `github.com/rs/cors` package.
+ */
+
+import go
+
+/**
+ * Provides abstract class for modeling the Go CORS handler model origin write.
+ */
+abstract class UniversalOriginWrite extends DataFlow::ExprNode {
+  abstract DataFlow::Node getBase();
+
+  abstract Variable getConfig();
+}
+
+/**
+ * Provides abstract class for modeling the Go CORS handler model allow all origins write.
+ */
+abstract class UniversalAllowAllOriginsWrite extends DataFlow::ExprNode {
+  abstract DataFlow::Node getBase();
+
+  abstract Variable getConfig();
+}
+
+/**
+ * Provides abstract class for modeling the Go CORS handler model allow credentials write.
+ */
+abstract class UniversalAllowCredentialsWrite extends DataFlow::ExprNode {
+  abstract DataFlow::Node getBase();
+
+  abstract Variable getConfig();
+}
+
+/**
+ * Provides classes for modeling the `github.com/rs/cors` package.
+ */
+module RsCors {
+  /** Gets the package name `github.com/gin-gonic/gin`. */
+  string packagePath() { result = package("github.com/rs/cors", "") }
+
+  /**
+   * A new function create a new rs Handler
+   */
+  class New extends Function {
+    New() { exists(Function f | f.hasQualifiedName(packagePath(), "New") | this = f) }
```

Comment:
```suggestion
    New() { this.hasQualifiedName(packagePath(), "New") }
```
`this` is already a `Function` because of the `extends Function` in the class declaration.